modules/ossm-config-otel.adoc

```diff
@@ -0,0 +1,142 @@
+* An Istio instance is created.
+* An Istio CNI instance is created.
+
+.Procedure
+
+. Navigate to the {OTELOperator} and install the `OpenTelemetryCollector` resource in the `istio-system` namespace:
++
+.Example OpenTelemetry Collector in `istio-system` namespace
+[source, yaml]
+----
+kind: OpenTelemetryCollector
+apiVersion: opentelemetry.io/v1beta1
+metadata:
+  name: otel
+  namespace: istio-system
+spec:
+  observability:
+    metrics: {}
+  deploymentUpdateStrategy: {}
+  config:
+    exporters:
+      otlp:
+        endpoint: 'tempo-sample-distributor.tempo.svc.cluster.local:4317'
+        tls:
+          insecure: true
+    receivers:
+      otlp:
+        protocols:
+          grpc:
+            endpoint: '0.0.0.0:4317'
+          http: {}
+    service:
+      pipelines:
+        traces:
+          exporters:
+            - otlp
+          receivers:
+            - otlp
+----
+
+. Configure {SMProductName} to enable tracing, and define the {OTELShortName} tracing providers in your `meshConfig`:
++
+.Example enabling tracing and defining tracing providers
+[source,yaml]
+----
+apiVersion: sailoperator.io/v1alpha1
+kind: Istio
+metadata:
+  ...
+  name: default
+spec:
+  namespace: istio-system
+  ...
+  values:
+    meshConfig:
+      enableTracing: true
+      extensionProviders:
+      - name: otel-tracing
+        opentelemetry:
+          port: 4317
+          service: otel-collector.istio-system.svc.cluster.local #<1>
+----
+<1> The `service` field is the `OpenTelemetry` collector service in the `istio-system` namespace.
+
+. Create an Istio Telemetry resource to enable tracers defined in `spec.values.meshConfig.ExtensionProviders`:
++
+.Example Istio Telemetry resource
+[source,yaml]
+----
+apiVersion: telemetry.istio.io/v1
+kind: Telemetry
+metadata:
+  name: otel-demo
+  namespace: istio-system
+spec:
+  tracing:
+    - providers:
+        - name: otel-tracing
+      randomSamplingPercentage: 100
+----
++
+[NOTE]
+====
+Once you verify that you can see traces, lower `randomSamplingPercentage` or set it to `default` to reduce the number of requests.
+====
+
+. Deploy `bookinfo`:
++
+[source, terminal]
+----
+oc create ns bookinfo
+oc label ns <namespace_name> istio.io/rev= # <1>
+oc apply -f https://raw.githubusercontent.com/istio/istio/release-1.23/samples/bookinfo/platform/kube/bookinfo.yaml -n bookinfo
+----
+<1> If you named your Istio resource `default` and are using the `InPlace`` upgrade strategy, use `oc label ns bookinfo istio-injection=enabled`.
+
+[NOTE]
+====
+To find your `<revision-name>`, run the following command:
+
+[source, terminal]
+----
+oc get istiorevisions.sailoperator.io
+----
+
+Sample output:
+
+[source,terminal]
+----
+NAME              TYPE    READY   STATUS    IN USE   VERSION   AGE
+default-v1-23-0   Local   True    Healthy   False    v1.23.0   3m33s
+----
+====
```

Comment:
Last nit: the indentation is off on this note. It doesn't match the surrounding blocks:
![Screenshot from 2024-10-14 08-45-07](https://github.com/user-attachments/assets/5b1ed52e-e000-4c96-8cfc-f9767550ecef)